diff --git a/service/http_client.go b/service/http_client.go
index c6febe06662b06c5d33992a13a78815a03c441d5..5a2986e2fefc30624e8c8774ce95cda748db7905 100644
--- a/service/http_client.go
+++ b/service/http_client.go
@@ -107,8 +107,12 @@ func newRelayHTTPClient(transport *http.Transport) *http.Client {
 }
 
 func InitHttpClient() {
+	dialer := &net.Dialer{}
 	transport := newRelayHTTPTransport()
 	transport.Proxy = http.ProxyFromEnvironment
+	transport.DialContext = func(ctx context.Context, network, addr string) (net.Conn, error) {
+		return dialer.DialContext(ctx, "tcp4", addr)
+	}
 	httpClient = newRelayHTTPClient(transport)
 	ssrfProtectedHTTPClient = newProtectedFetchHTTPClient()
 }
